### DISCUSSION TITLE
Is it possible to not show the current page breadcrumb as a hyperlink?

### DISCUSSION THREAD

**Discussion by glow12121** [datenstrom/community#506] [Ask a question] [answered]
Hi,

Firstly, many thanks to all the makers of Yellow and its extensions! It is a beautiful engine for websites.

Now, I was wondering if it is possible to get the [Breadcrumbs plugin](https://github.com/datenstrom/yellow-extensions/tree/master/features/breadcrumb)  not to show the current page breadcrumb as a hyperlink.

For example, this is how it is now:

[Home](HTTP://home.org) > [Rubric](HTTP://home.org/rubric/) > [Article](HTTP://home.org/rubric/article/)

And here is how I would like it to be:

[Home](HTTP://home.org) > [Rubric](HTTP://home.org/rubric/) > Article

No hyperlink for the page where you are now - is it possible? Thanks.

**Comment by GiovanniSalmeri** [comment, marked as answer]
It should be sufficient to change the lines 28-31 in breadcrumb.php with the following ones:

```php
foreach ($pages as $pageBreadcrumb) {
    if ($pageBreadcrumb->getLocation(true)!=$page->getLocation(true))
        $output .= "<a href=\"".$pageBreadcrumb->getLocation(true)."\">".$pageBreadcrumb->getHtml("titleNavigation")."</a> ".htmlspecialchars($separator)." ";
    else
        $output .= $pageBreadcrumb->getHtml("titleNavigation");
}
```

(Not tested!) In fact, according to [Jakob Nielsen](https://www.nngroup.com/articles/breadcrumbs/) this should be the correct behaviour ("In the breadcrumb trail, the breadcrumb corresponding to the current page should not be a link.") @markseu, do you have any opinion?

**Comment by glow12121** [comment]
:)

![Screenshot from 2020-04-20 10-05-38](https://user-images.githubusercontent.com/29035180/79723656-ca889f00-82ee-11ea-8cd8-b05b2012768e.png)

**Comment by markseuffert** [comment]
Let's follow Jakob Nielsen's advice. Go ahead, find a working solution and send a pull request.

**Comment by glow12121** [comment]
Well, I've tested Breadcrumbs on my site today and it seems to be not working. For example, this is the index page's code:

```
---
Title: Main
---

[image picture.jpg]

Text text text

[breadcrumb]

[breadcrumb > breadcrumb]

```

After having been `built` and `served`, the outcome page just shows these lines, literally:

![Screenshot from 2020-04-20 19-49-58](https://user-images.githubusercontent.com/29035180/79777653-3cd59f80-8340-11ea-997b-b83e5027961f.png)

In `system.ini`, the settings for Breadcrumbs are these:

```
BreadcrumbsSeparator: |
BreadcrumbsStyle: breadcrumbs
```

**Comment by GiovanniSalmeri** [comment]
I've just tested my modification and for me it works as intended...Your output is as if the extension breadcrumb were not at all installed. Is the file `breadcrumb.php` (*exactly with this name*) in the folder `system/extensions`?

**Comment by glow12121** [comment]
Thanks,  Giovanni! For some reason, I had `breadcrumbs.php`, in the plural, in the extensions folder, as well as in all corresponding settings.

I re-installed this plugin and: by default, it had the active breadcrumb with a hyperlink. But after I amended the PHP file with the code you suggested, the hyperlink, indeed, disappeared.

So, what's next? I can assume that after the next update these lines will be deleted from the PHP file.

**Comment by GiovanniSalmeri** [comment]
Since the core developers agree, I will make as soon as possible a pull request in order to modify the standard extension of Yellow. So the next update will take to all the users the correction you have suggested.

**Comment by glow12121** [comment]
Once again, many thanks!

**Comment by glow12121** [comment]
One more question if I may: is it possible to get the plugin not to show the opened page as a breadcrumb at all? Like this:

`Yellow | Issues |`

Instead of:

`Yellow | Issues | Breadcrumb`

**Comment by GiovanniSalmeri** [comment]
Very easy, delete these two lines:

```
    else
        $output .= $pageBreadcrumb->getHtml("titleNavigation");
```

**Comment by BsNoSi** [comment]
@thumos333 : Yellow typically takes care of modified extensions. You can check the individual behaviour of each file by checking the `extension.ini` file. If there is a `careful` behind an install path this file is not overwritten, you have to force it.

To *freeze* changes of a modified extensions without »careful« you can take the simple way by renaming it. »[mybreadcumb]« will never be overwritten by »[breadcumb]« – the latter is redundent then but the first will not be maintained by the community.

Therefore, this way (discussing your ideas here) is the better one. Sharing your ideas may not lead to the result you specifically expect. But good ideas will go into the development of an extension or the whole system. This is granted. And this improves YELLOW for all users.

Besides of that you are welcome to offer your *coded solutions* here. (As you can see) You will be treated with respect an you will get help from those who can help. Every question/answer improves the knowledge of the whole community.

**Comment by glow12121** [comment]
Thanks, @GiovanniSalmeri !

@BsNoSi , I only know basic html/css and reporting inconsistencies or doing testing is the best I can do to help. This is all I've got in `extensions.ini`, no plugins are mentioned:

```
# Datenstrom Yellow extension

Extension: Core
Version: 0.8.3
Description: Core functionality for your website.
Published: 2019-04-05 14:22:01
Developer: Datenstrom

Core/core.php: system/extensions/core.php,create,update
```

**Comment by BsNoSi** [comment]
See last line: »core.php« is the extension ("plugin") that will be »create«(ed) or »update«(d) when you copy the zip-file of the extension into the system→extensions folder.


This means: core.php will be *always* updated. Which is - in this case – necessary to get the latest general system changes that usually offer new basic functionalities or fix bugs. "core" ist the "engine" of YELLOW other extensions are "tuning parts".

*Any* help is welcome. Finding inconsitencies is the hardest job of all…

**Comment by glow12121** [comment]
@BsNoSi , okay, thanks. So in order to prevent Yellow from updating a particular extension, I need to add lines manually here, right? The thing is I already have quite a few extensions in the `extensions` folder. But they aren't mentioned in `extensions.ini`.

**Comment by BsNoSi** [comment]
Each extension has its own "extension.ini" that is deleted after unpacking the zip. This is a basic process of the system to prevent mix up of actions.

#### Update Rules

- **Always create a backup** of your *whole* system **before updating**.
- The update function of YELLOW includes only "official" extensions (offered from Datenstrom). All supplemental from others have to be updated "by hand" (copy latest version into extension folder).
- "Official" extensions update (as far as I know) always *all* php parts. CSS/html parts are only updated if the existing are untouched (→not changed).
   - This overrides changes in translation files, too. → s. "backup" notice above…
- Extensions from other authors *should* behave the same but sometimes they don't → see readme for details *before* updating. If not this should be mentioned an has reasons, e.g. that the new version needs functionality of/in new html/css/js files.

**Comment by schulle4u** [comment]
>     * The update function of YELLOW includes only "official" extensions (offered from Datenstrom). All supplemental from others have to be updated "by hand" (copy latest version into extension folder).

All extensions can be distributed as updatable website feature in the official Datenstrom repository, as long as they follow the coding and documentation standards. I suggest to have a look at the [release extension](https://github.com/datenstrom/yellow-extensions/tree/master/features/release), the documentation shows you how to prepare an extension for the extensions repository.

**Comment by annaesvensson** [comment]
Thanks for your feedback. I am closing this issue,  extension has been updated as suggested.

**Comment by glow12121** [comment]
> Thanks for your feedback. I am closing this issue, extension has been updated as suggested.

Thank you for helping!
